### DISCUSSION TITLE
Where does WinGet install portable packages?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [microsoft/winget-pkgs#184459] [Q&A] [answered]
I've installed [`marticliment.com/unigetui/share?id=GodotEngine.GodotEngine.Mono&sourceName=winget`](https://marticliment.com/unigetui/share?name=Godot+Engine+(Mono)&id=GodotEngine.GodotEngine.Mono&sourceName=winget&managerName=WinGet). However, because of https://github.com/microsoft/winget-cli/issues/2299#issue-1291406635, I have no idea of how to invoke it.

**Comment by Trenly** [comment]
You can use `winget --info` to list out where the portable packages are installed on your machine. Where each specific package is installed will depend on if it is x86 or x64, and if it was installed with machine or user scope

**Reply by roke-julian-lockhart** [reply]
> https://github.com/microsoft/winget-pkgs/discussions/184459#discussioncomment-10993813

@Trenly, `winget` has no `--info` flag for me, and https://github.com/microsoft/winget-cli/issues/2298#issue-1291319187 appears to indicate that this functionality doesn't actually exist...

*	1.	```PS1
		#!/usr/bin/env -S pwsh
		#Requires -PSEdition 'Core'
		#Requires -Version 7.5
		If ($IsWindows) { winget --info 'GodotEngine.GodotEngine.Mono' }
		```
		
	1.	> ```YAML
		> Found a positional argument when none was expected: 'GodotEngine.GodotEngine.Mono'
		> ```
	

*	1.	```PS1
		#!/usr/bin/env -S pwsh
		#Requires -PSEdition 'Core'
		#Requires -Version 7.5
		If ($IsWindows) { winget --show 'GodotEngine.GodotEngine.Mono' }
		```
		
	1.	> ```YAML
		> Argument name was not recognized for the current command: '--show'
		> ```

**Reply by roke-julian-lockhart** [reply, marked as answer]
> https://github.com/microsoft/winget-pkgs/discussions/184459#discussioncomment-10996597

Apologies, @Trenly, for I should have read the error properly, for merely `winget --info` works:

1.	~~~pwsh
	#!/usr/bin/env -S pwsh
	#Requires -PSEdition 'Core'
	#Requires -Version 7.5
	If ($IsWindows) { winget --info }
	~~~

1.	<blockQuote cite="file:///C:/Users/rokej/AppData/Local/Microsoft/WindowsApps/winget.exe">

	```YAML
	PS C:\Users\rokej> winget --info
	Windows Package Manager (Preview) v1.9.2411-preview
	Copyright (c) Microsoft Corporation. All rights reserved.

	Windows: Windows.Desktop v10.0.26100.2152
	System Architecture: X64
	Package: Microsoft.DesktopAppInstaller v1.24.2411.0

	Winget Directories
	-------------------------------------------------------------------------------------------------------------------------------
	Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
	User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
	Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
	Portable Links Directory (Machine) C:\Program Files\WinGet\Links
	Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
	Portable Package Root              C:\Program Files\WinGet\Packages
	Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
	Installer Downloads                %USERPROFILE%\Downloads
	```

	</blockQuote>

#### **Examples**

Per that, it was trivial to ascertain that Godot was installed into:

~~~pwsh
#!/usr/bin/env -S pwsh
#Requires -PSEdition 'Core'
#Requires -Version 7.5
If ($IsWindows) { Start-Process "$Env:SystemDrive\Users\$([Environment]::UserName)\AppData\Local\Microsoft\WinGet\Packages\GodotEngine.GodotEngine.Mono_Microsoft.Winget.Source_8wekyb3d8bbwe" }
~~~

Likewise, for the Media Creation Tool:

~~~pwsh
#!/usr/bin/env -S pwsh
#Requires -PSEdition 'Core'
#Requires -Version 7.5
If ($IsWindows) { Start-Process "$Env:SystemDrive\Users\$([Environment]::UserName)\AppData\Local\Microsoft\WinGet\Packages\Microsoft.MediaCreationTool_Microsoft.Winget.Source_8wekyb3d8bbwe" }
~~~

It would have been nice if there were a PowerShell commandlet for this, though. Perhaps I'll request one. Got any ideas for a name?
